javacutil/src/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -1017,4 +1046,41 @@ public static boolean isArrayLengthAccess(Tree tree) {
         }
         return false;
     }
+
+    /**
+     * finds the corresponding functional interface method for a lambda expression
+     *
+     * @param tree the lambda expression
+     * @return the functional interface method
+     */
+    public static Symbol.MethodSymbol getFunctionalInterfaceMethod(LambdaExpressionTree tree) {
+        Type funcInterfaceType = ((JCTree.JCLambda) tree).type;
```

Comment:
There is, actually. We found out recently that this can be done through the `com.sun.tools.javac.code.Types.findDescriptorSymbol` API, updating this PR accordingly.